### PR TITLE
For Comment: Load Converters and Config via annotations

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**
Example of how we can load things via annotations and not have to worry about META-INF/resources SPI files...Users can just annotate things. We could also do a SimpleFeatureType loading this way as well.

Lemme know what you think.

Signed-off-by: Andrew Hulbert <andrew.hulbert@ccri.com>

### CODE REVIEW

**Comment by jnh5y** [comment]
@jahhulbert-ccri missing headers ahoy!
[WARNING] Missing header in: /home/travis/build/locationtech/geomesa/geomesa-convert/geomesa-convert-common/src/main/java/org/locationtech/geomesa/convert/annotations/ConverterConfig.java

[WARNING] Missing header in: /home/travis/build/locationtech/geomesa/geomesa-convert/geomesa-convert-common/src/main/java/org/locationtech/geomesa/convert/annotations/ConverterFactory.java

[WARNING] Missing header in: /home/travis/build/locationtech/geomesa/geomesa-convert/geomesa-convert-common/src/main/java/org/locationtech/geomesa/convert/annotations/TransformFactory.java

File: pom.xml [hunk 1]
```diff
@@ -1876,6 +1876,11 @@
                 <artifactId>jcommander</artifactId>
                 <version>1.48</version>
             </dependency>
+            <dependency>
+                <groupId>org.reflections</groupId>
+                <artifactId>reflections</artifactId>
+                <version>0.9.10</version>
```
Comment (jnh5y) [inline pom.xml]: What kinda license we looking at?

**Comment by jnh5y** [comment]
The nice thing about SPI is that editing a text file can control what is available.  Is there a good way to setup annotations after the fact?  I.e., if we miss an annotation, we'd have to recompile, right?

Other than preference of advertisement, are there other reasons to consider annotations?  

I think I'd need a little convincing...

**Comment by jahhulbert-ccri** [comment]
closing. not doing it looks like.

**Pull request closed without merging**